cmd/terramate/e2etests/list_test.go

```diff
@@ -195,6 +194,90 @@ terramate {
 	assertRunResult(t, cli.listChangedStacks(), want)
 }
 
+func TestListChangedIgnoreDeletedStackDirectory(t *testing.T) {
+	s := sandbox.New(t)
+
+	stack := s.CreateStack("stack-old")
+	cli := newCLI(t, s.RootDir())
+
+	git := s.Git()
+	git.CommitAll("all")
+	git.Push("main")
+	git.CheckoutNew("deleted-stack")
+
+	test.RemoveAll(t, stack.Path())
+
+	git.CommitAll("removed stack")
+
+	assertRun(t, cli.listChangedStacks())
+}
+
+func TestListChangedIgnoreDeletedNonStackDirectory(t *testing.T) {
+	s := sandbox.New(t)
+
+	s.CreateStack("stack")
+	toBeDeletedDir := filepath.Join(s.RootDir(), "to-be-deleted")
+	test.MkdirAll(t, toBeDeletedDir)
+	test.WriteFile(t, toBeDeletedDir, "test.txt", "")
+	cli := newCLI(t, s.RootDir())
+
+	git := s.Git()
+	git.CommitAll("all")
+	git.Push("main")
+
+	git.CheckoutNew("deleted-diretory")
+
+	test.RemoveAll(t, toBeDeletedDir)
+	git.CommitAll("removed directory")
+
+	assertRun(t, cli.listChangedStacks())
+}
+
+func TestListChangedDontIgnoreStackDeletedFiles(t *testing.T) {
+	s := sandbox.New(t)
+
+	stack := s.CreateStack("stack")
+	testDir := stack.CreateDir("test")
+	file := testDir.CreateFile("testfile", "")
+	cli := newCLI(t, s.RootDir())
+
+	git := s.Git()
+	git.CommitAll("all")
+	git.Push("main")
+	git.CheckoutNew("deleted-file")
+
+	test.RemoveAll(t, file.Path())
+
+	git.CommitAll("removed file")
+
+	assertRunResult(t, cli.listChangedStacks(), runExpected{
+		Stdout: stack.RelPath() + "\n",
+	})
+}
+
+func TestListChangedDontIgnoreStackDeletedDirectories(t *testing.T) {
+	s := sandbox.New(t)
+
+	stack := s.CreateStack("stack")
+	testDir := stack.CreateDir("test")
+	testDir.CreateFile("testfile1", "")
+	testDir.CreateFile("testfile2", "")
```

Comment:
```suggestion
	testDir.CreateFile("testfile", "")
```

Doesnt seem we need 2 files to test this, but disregard if we need.

Reply:
as git knows nothing about directories, only files, at the git level, this function tests if we detect the deletion of multiple files inside subdirectories of the stack, so if we don't add multiple files it doesn't look like a directory deletion.
Also this tests that we ignore other files if the detect was already marked as changed.